x/blob/client/cli/payforblob.go

```diff
@@ -63,11 +65,20 @@ func CmdPayForBlob() *cobra.Command {
 	}
 
 	flags.AddTxFlagsToCmd(cmd)
+	cmd.PersistentFlags().Uint8(FlagNamespaceVersion, 0, "Specify the namespace version")
 	cmd.PersistentFlags().Uint8(FlagShareVersion, 0, "Specify the share version")
-
 	return cmd
 }
 
+func getNamespace(namespaceID []byte, namespaceVersion uint8) (appns.Namespace, error) {
+	switch namespaceVersion {
+	case appns.NamespaceVersionZero:
+		return appns.New(namespaceVersion, append(appns.NamespaceVersionZeroPrefix, namespaceID...))
```

Comment:
we should never append directly to a global, as it could cause tons of bugs

we need to first make an empty slice of appropriate capacity, and then append

Reply:
I don't think we are appending to global value. Append function will return a value here which is passed to `appns.New()`
z = append(x, y)
This line won't change x or y.

Reply:
My takeaway after reading that article is that 

```go
z = append(x, y)
```

is safe because the length or capacity of x will not be impacted by the above statement. After execution the global `x` remains unmodified. My understanding is that the copy is only necessary if a function is invoked on the global `x` that modifies elements of the original slice. If `append` is dangerous in this usage, I would expect there to be a linter to warn us about incorrect usage but I couldn't find one on https://golangci-lint.run/usage/linters/

@evan-forbes how do we enforce that `append` isn't misused? Additionally are there examples of just the `append` causing bugs:

> we should never append directly to a global, as it could cause tons of bugs

Reply:
I think we have some linter, but otherwise this is just good practice imo since it always fixes the issue and eliminates the reader having to think "is this an issue". Also, I still think its an issue but don't have the brain _capacity_ at the moment to figure it out lol :laughing: 

examples https://github.com/celestiaorg/celestia-core/issues/336, https://github.com/celestiaorg/celestia-core/issues/250, https://github.com/celestiaorg/nmt/pull/30, there was also one in node that fixed during the barcelona onsite but I couldn't find it in a cursory search

**all** were caused by 
z = append(x, y)

so now we should never ever ever do that no matter what because evan is too scarred